test/parallel/test-debug-port-numbers.js

```diff
@@ -44,7 +44,10 @@ function kill(child) {
   try {
     process.kill(-child.pid);  // Kill process group.
   } catch (e) {
-    assert.strictEqual(e.code, 'ESRCH');  // Already gone.
+    if (process.platform === 'darwin')
+      assert.strictEqual(e.code, 'EPERM');  // Already gone.
+    else
+      assert.strictEqual(e.code, 'ESRCH');  // Already gone.
```

Comment:
I think the logic should be ESRCH || EPERM iff OS X, and you might as well drop the 'iff OS X' part in that case because the BSDs probably exhibit the same behavior.

I checked the xnu and libc sources and the error code seems to depend on whether POSIX compatibility mode is enabled in the libc wrapper: it returns EPERM if it is, ESRCH otherwise.  That would explain why I don't see EPERM with 10.8.